### ISSUE TITLE
Tracing alias instructions changes their behaviour

### ISSUE THREAD

**Issue by lucat1** [jolie/jolie#460]
**Describe the bug**
When using `--trace` to interpret a piece of code which uses aliasing, the behaviour differs from the standard execution mode. When aliasing `gloal.test.(k)` the node at `global.test.(k).(k)` gets created. This doesn't happen without tracing. When writing, the behaviour is as expected.

**To Reproduce**
Run the following snippet of code with or without `--trace`.

```jolie
from console import Console
from string_utils import StringUtils

service Main {
  embed Console as Console
  embed StringUtils as StringUtils

  main {
    key = "k"
    valueToPrettyString@StringUtils(global)(t)
    println@Console("before aliasing: " + t )()

    test -> global.test.(key)

    valueToPrettyString@StringUtils(global)(t)
    println@Console("after aliasing: " + t )()

    test = 1

    valueToPrettyString@StringUtils(global)(t)
    println@Console("after write: " + t )()
  }
}
```

**Expected behavior**
The tracing behaviour should match the stadard one.

**Desktop (please complete the following information):**
 - OS: Linux (both Docker and native)
 - Jolie version: lastest master or docker's `alpine-edge`
 - Java Version: 20.0.1

**Comment by fmontesi** [comment]
Yeah, looks like the quick solution for now would be not to print the target. What's the output if you do that?

**Comment by lucat1** [comment]
If I remove the call I mentioned previously or the whole `tracer.trace` the program behaves as expected.

**Comment by mwallnoefer** [comment]
> I have debugged this a bit and the issue goes away if i replace this line with `null`:
> 
> [jolie/jolie/src/main/java/jolie/process/MakePointerProcess.java](https://github.com/jolie/jolie/blob/4ef16d92a9496c79e187332e954cdb4a57ab049e/jolie/src/main/java/jolie/process/MakePointerProcess.java#L70)
> 
> Line 70 in [4ef16d9](/jolie/jolie/commit/4ef16d92a9496c79e187332e954cdb4a57ab049e)
> 
>  rightPath.getValue( ExecutionThread.currentThread().state().root().clone() ),

@fmontesi , @lucat1, @Niels-Erik couldn't we fix this by substituting `rightPath` by `leftPath` in this specific line (since they are. just aliases)?

**Comment by Niels-Erik** [comment]
> [@fmontesi](https://github.com/fmontesi) , [@lucat1](https://github.com/lucat1), [@Niels-Erik](https://github.com/Niels-Erik) couldn't we fix this by substituting `rightPath` by `leftPath` in this specific line (since they are. just aliases)?

Then we would run into the exact same problem as in #562.

**Comment by klag** [comment]
Is it still open?

**Comment by Niels-Erik** [comment]
> Is it still open?

yes

**Comment by Niels-Erik** [comment]
I had a look at this, and the problem arises because the aliasing operator does not normally evaluate but the the logging makes it do, which is a problem with paths that has side effect in them (like this problem or "i++" syntax). 

So I see two choices:
1. Do not log the evaluated value on aliasing (easy).
2. Find a way to either revert the side effects afterwards or make the side effect not take place.

@klag @fmontesi @mwallnoefer thoughts?
